benchmarks/inspector_benchmarks.cpp

```diff
@@ -0,0 +1,49 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <cassert>
+#include <memory>
+
+#include "benchmark/benchmark.h"
+
+#include "umpire/config.hpp"
+
+#include "umpire/strategy/mixins/Inspector.hpp"
+#include "umpire/strategy/AllocationStrategy.hpp"
+
+static void benchmark_inspector(benchmark::State& state) {
+  auto inspector = umpire::strategy::mixins::Inspector();
+
+  double* data = new double[256];
+
+  void* ptr = static_cast<void*>(data);
+  size_t length = 256*sizeof(double);
+  std::shared_ptr<umpire::strategy::AllocationStrategy> strat(nullptr);
+
+  while (state.KeepRunning()) {
+    inspector.registerAllocation(
+        ptr,
```

Comment:
What happens when the same pointer is given to registerAllocation again and again?  Does it cause new records to be created for each registration?

Reply:
A new record will be created (from the pool), and it will be inserted at the end of the vector in the AllocationMap

Reply:
Ah yes, I believe I remember this now.  It carries with it an implied assumption that deallocations will be done in reverse order from allocations and it is necessary because this is what will allow pools to work where they may have one allocation for the block and a different one for a sub-allocation of the block.